pygmt/clib/conversion.py

```diff
@@ -102,6 +104,11 @@ def dataarray_to_matrix(grid):
             )
         region.extend([coord.min(), coord.max()])
         inc.append(coord_inc)
+
+    if any([i < 0 for i in inc]):  # Sort grid when there are negative increments
+        inc = [abs(i) for i in inc]
+        grid = grid.sortby(variables=list(grid.dims), ascending=True)
```

Comment:
One thing to check here is if this generates a memory copy. Though there isn't a good way around it, I suppose, since `as_c_contiguous` will have to make a copy if that's the case.

Reply:
I believe it does (create a memory copy). Maybe we should revert back to the if-then checks then? Not as readable but if speed is what matters then it'll be faster.

Alternatively, we *could* require users to sort their grid properly (be it via flipping the axes or using sortby) before passing into PyGMT, but it won't be a fantastic user experience.

Reply:
I think that this is probably ok as is: If the grid is properly ordered, then this code block doesn't get executed anyways.